### DISCUSSION TITLE
After changing Webpack5, the build time is very slow.

### DISCUSSION THREAD

**Discussion by kdevkr** [webpack/webpack#15161] [Q&A] [answered]
In Node v14, macOS BigSur

**Webpack4**
- webpack@4.46.0
- webpack-cli@3.3.12
- webpack-dev-server@3.11.2
- babel/core@7.15.8
- babel-loader@8.2.3
- css-loader@5.2.7
- sass@1.32.13
- sass-loader@9.0.3
- sass-resources-loader@2.2.4
- postcss-loader@4.3.0
- resolve-url-loader@3.1.4
- terser-webpack-plugin@4.2.3
- vue-loader@15.9.8
- vue-style-loader@4.1.3

**Webpack5**
- webpack@5.65.0
- webpack-cli@4.9.1
- webpack-dev-server@4.7.2
- terser-webpack-plugin@5.3.0
- css-minimizer-webpack-plugin@3.3.1
- css-loader@6.5.1
- postcss-loader@6.2.1
- sass@1.45.2
- sass-loader@12.4.0

**Performance**
- webpack4: 64399ms (dev-server: 89409ms)
- webpack5: 186518ms (dev-server:135508ms), _Webpack5 no filesystem cache for compare._

According to the guide, the loader and plug-in were changed to the latest version, file-loader to asset-module, and excluded node-polyfill was added. (path, stream, buffer, util, process). 
Is it a normal result to increase the build time like this?

**Comment by alexander-akait** [comment, marked as answer]
Will be great to look at configuration, I think something wrong with setup

**Comment by alexander-akait** [comment]
Is problem only in production mode or development?

**Comment by sokra** [comment]
Difficult to say without profile. My guess would be that it's somehow related to sass.

You can set `profile: true` option in webpack 5 to get more timings of module build times

**Comment by kdevkr** [comment]
![image](https://user-images.githubusercontent.com/17937604/149430591-cf33b814-cbb0-4d3c-a1c1-5f3d43ddfa5f.png)

I check slow of each loader and plugin. however, the versions are diff. so, accurate comparison seems impossible.

![image](https://user-images.githubusercontent.com/17937604/149433053-65ae5fa1-5659-4bd0-bf13-5ee4c1a02256.png)
when rebuilding, Not files in the `node_modules/.cache` folder of `terser-webpack-plugin` and `css-minimizer-webpack-plugin`. is there any update related to this?...  has the default value changed due to filesystem cache support?

**Reply by alexander-akait** [reply]
You can setup cache using `cache: { type: "filesystem" }` in `webpack.config.js`, also I think you don't need some loaders, like `resolver-url-loader`, most of CSS frameworks/libraries allow to set path to assets, `resolver-url-loader` parse CSS, it is very expensive operation

**Reply by kdevkr** [reply]
If Webpack 5 does not specify a file system cache, should I understand that it is not cached for plugins such as terser or css-minimizer provided by Webpack?. however, I don't think we can find any related information in the webpack or each plugin repos.
resolve-url-loader is also used in webpack 4, so i will review the overhead you mentioned separately.

**Reply by alexander-akait** [reply]
> If Webpack 5 does not specify a file system cache, should I understand that it is not cached for plugins such as terser or css-minimizer provided by Webpack?

Yes, if you need memory/file system cache, you need set it on webpack.config.js level, for webpack v4 each plugins has own cache system, now they are using webpack API and configured using webpack
